src/main/python/e2e_experimental_settings.py

```diff
@@ -5,23 +5,23 @@
 import shlex
 import subprocess
 
+
 class Setting(object):
     def __init__(self, label, value_flag_map):
         self.label = label
         self.choice_flags = value_flag_map
-    
+
     def get_settings(self):
         return self.choice_flags.keys()
 
     def get_choice(self, setting):
         return self.choice_flags[setting]
 
     def get_options(self):
-        options = []
-        for key in self.choice_flags.keys():
-            options.append("{}:{}".format(self.label, key))
+        options = [options.append("{}:{}".format(self.label, key)) for key in self.choice_flags.keys()]
```

Comment:
Did you check that this works? I don't think you did.

`list.append` returns `None`, which would mean that `options` ends up as `[None, None,...]`. Also, given that you're just then going to return the list, I'd suggest replacing by:
`return ["{}:{}".format(self.label, key) for key in self.choice_flags.keys()]`
Also if this doesn't need to be a list, then changing to a generator (by swapping the outermost `[]` with `()`) will be marginally more efficient